### PR TITLE
feat(blockstore): add IndexedDb blockstore

### PR DESCRIPTION
**Pull request by zvolin**

### CODE REVIEW

File: blockstore/src/indexed_db_blockstore.rs [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+use rexie::{KeyRange, ObjectStore, Rexie, Store, TransactionMode};
+use send_wrapper::SendWrapper;
+use wasm_bindgen::{JsCast, JsValue};
+
+use crate::{convert_cid, Blockstore, BlockstoreError, Result};
+
+/// indexeddb version, needs to be incremented on every schema schange
+const DB_VERSION: u32 = 1;
+
+const BLOCKS_STORE: &str = "BLOCKSTORE.BLOCKS";
```
Comment (fl0rek) [inline blockstore/src/indexed_db_blockstore.rs:12]:
```suggestion
const BLOCK_STORE: &str = "BLOCKSTORE.BLOCKS";
```

File: blockstore/src/indexed_db_blockstore.rs [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    /// let blockstore = IndexedDbBlockstore::<64>::new(db).await?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub async fn new(name: &str) -> Result<Self> {
+        let rexie = Rexie::builder(name)
+            .version(DB_VERSION)
+            .add_object_store(
+                ObjectStore::new(BLOCKS_STORE)
+                    // .key_path("cid")
```
Comment (fl0rek) [inline blockstore/src/indexed_db_blockstore.rs:39]:
```suggestion
```

File: blockstore/src/indexed_db_blockstore.rs (hunk 1 above)
Comment (zvolin) [reply blockstore/src/indexed_db_blockstore.rs:12]: ah this one was actually intentional :D a store for blocks. I can change that if you want

Comment (fl0rek) [reply blockstore/src/indexed_db_blockstore.rs:12]: 🤔 wouldn't that still be singular? Like "flower shop" or "grain store"? fwiw google is calling their api for storing and retrieving blocks of data (for credential purposes) [Block Store](https://developers.google.com/identity/blockstore/android).
Where's native speaker when you need one 😿

File: blockstore/src/indexed_db_blockstore.rs [hunk 3]
```diff
@@ -0,0 +1,160 @@
+use cid::CidGeneric;
+use js_sys::Uint8Array;
+use rexie::{KeyRange, ObjectStore, Rexie, Store, TransactionMode};
+use send_wrapper::SendWrapper;
+use wasm_bindgen::{JsCast, JsValue};
+
+use crate::{convert_cid, Blockstore, BlockstoreError, Result};
+
+/// indexeddb version, needs to be incremented on every schema schange
```
Comment (fl0rek) [inline blockstore/src/indexed_db_blockstore.rs:9]:
```suggestion
/// indexeddb version, needs to be incremented on every schema change
```

**Review by fl0rek — APPROVED** [review]
🎸

File: blockstore/src/indexed_db_blockstore.rs [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+            .map_err(|e| BlockstoreError::BackingStoreError(e.to_string()))?;
+
+        Ok(Self {
+            db: SendWrapper::new(rexie),
+        })
+    }
+
+    async fn get<const S: usize>(&self, cid: &CidGeneric<S>) -> Result<Option<Vec<u8>>> {
+        let cid = convert_cid::<S, MAX_MULTIHASH_SIZE>(cid)?;
+        let cid = Uint8Array::from(cid.to_bytes().as_ref());
```
Comment (oblique) [inline blockstore/src/indexed_db_blockstore.rs:47]: Do you use the `convert_cid` as a validation mechanism? Because otherwise it is not really needed.

Comment (zvolin) [reply blockstore/src/indexed_db_blockstore.rs]: removed the MAX_MULTIHASH_SIZE from sled and indexeddb blockstores

**Review by oblique — APPROVED** [review]
lgtm

**Review by fl0rek — APPROVED** [review]
Nice, cool test changes  👌

**Pull request merged**